proof/aggregate.saw

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (c) 2020 Galois, Inc.
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+*/
+
+///////////////////////////////////////////////////////////////////////////////
+// Specifications
+///////////////////////////////////////////////////////////////////////////////
+
+let blst_pairing_init_spec DST_len = do {
+  ctx_ptr <- crucible_alloc (llvm_struct "struct.PAIRING");
+  hash_or_encode <- crucible_fresh_var "hash_or_encode" (llvm_int 32);
+  //(DST, DST_ptr) <- ptr_to_fresh_readonly "DST" (llvm_array DST_len (llvm_int 8)); // not strictly needed for this function
+  DST_ptr <- crucible_fresh_pointer (llvm_array DST_len (llvm_int 8));
+  crucible_execute_func [ctx_ptr, crucible_term hash_or_encode, DST_ptr, crucible_term {{`DST_len:[64]}}];
+  ctx_ctrl <- crucible_fresh_var "new_pairing_init_ctx_ctrl" (llvm_int 32);
+  crucible_points_to (crucible_field ctx_ptr "ctrl") (crucible_term ctx_ctrl);
+  crucible_points_to (crucible_field ctx_ptr "nelems") (crucible_term {{0:[32]}});
+  crucible_points_to (crucible_field ctx_ptr "DST") DST_ptr;
+  crucible_points_to (crucible_field ctx_ptr "DST_len") (crucible_term {{`DST_len:[64]}});
+};
+
+// the enum for aggregate-operations errors
+let AGGR_UNDEFINED = {{ zero:[32] }};
+let AGGR_MIN_SIG = {{ (zext 0x1):[32] }};
+let AGGR_MIN_PK = {{ (zext 0x2):[32] }};
+let AGGR_SIGN_SET = {{ (zext 0x10):[32] }};
+let AGGR_GT_SET = {{ (zext 0x20):[32] }};
+let AGGR_HASH_OR_ENCODE = {{ (zext 0x40):[32] }};
+let MIN_SIG_OR_PK = {{ AGGR_MIN_SIG || AGGR_MIN_PK }};
+
+// BLST_ERROR enum:
+let BLST_SUCCESS = 0;
+let BLST_AGGR_TYPE_MISMATCH = 4;
+let BLST_PK_IS_INFINITY = 6;
+let BLST_POINT_NOT_IN_GROUP = 3;
+
+let N_MAX = 8; // NOTE: must match N_MAX defined in C code
+
+let blst_pairing_aggregate_pk_in_g1_spec DST_len msg_len aug_len nelems null_sig = do {
```

Comment:
can this be combined with `blst_pairing_aggregate_pk_in_g2_spec`?

Reply:
Yes it probably should!